### PR TITLE
Port boops and handshakes

### PR DESCRIPTION
**Pull request by r4d6**
## About The Pull Request

This port handshakes and nose booping from CHOMPStation

https://github.com/CHOMPstation/CHOMPstation/pull/638

## Why It's Good For The Game

I ported the handshakes while porting the nose booping while I was at it.

For the nose booping, do I really need to explain?

## Changelog
:cl:
add: Added Handshakes
add: Added Nose booping
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -316,7 +316,17 @@
...
 			else
 				return
-
+		
+		else if(H.zone_sel.selecting == "r_hand" || H.zone_sel.selecting == "l_hand")
+			H.visible_message( \
+				"<span class='notice'>[H] shakes [target]'s hand.</span>", \
+				"<span class='notice'>You shake [target]'s hand.</span>", )
+			
+		else if(H.zone_sel.selecting == "mouth") // I ADDED BOOP-EH-DEH-NOSEH - Jon
```
Comment (deathride58) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
		else if(check_zone(M.zone_selected) == "mouth") // I ADDED BOOP-EH-DEH-NOSEH - Jon
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 2]
```diff
@@ -316,7 +316,17 @@
 
 			else
 				return
-
+		
+		else if(H.zone_sel.selecting == "r_hand" || H.zone_sel.selecting == "l_hand")
```
Comment (deathride58) [inline code/modules/mob/living/carbon/carbon_defense.dm]: we dont have hands
```suggestion
		else if(check_zone(M.zone_selected) == "r_arm" || check_zone(M.zone_selected) == "l_arm")
```

**Review by deathride58 — CHANGES_REQUESTED** [review]

**Comment by deathride58** [comment]
bonus points if you add a tiny lil plush squeak noise when the boop occurs

**Comment by r4d6** [comment]
Done

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 3]
```diff
@@ -316,7 +316,18 @@
 
 			else
 				return
-
+		
+		else if(check_zone(M.zone_selected) == "r_arm" || check_zone(M.zone_selected) == "l_arm")
+			H.visible_message( \
```
Comment (deathride58) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
			M.visible_message( \
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 4]
```diff
@@ -316,7 +316,18 @@
 
 			else
 				return
-
+		
+		else if(check_zone(M.zone_selected) == "r_arm" || check_zone(M.zone_selected) == "l_arm")
+			H.visible_message( \
+				"<span class='notice'>[H] shakes [target]'s hand.</span>", \
```
Comment (deathride58) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
				"<span class='notice'>[M] shakes [src]'s hand.</span>", \
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 5]
```diff
@@ -316,7 +316,18 @@
...
 				return
-
+		
+		else if(check_zone(M.zone_selected) == "r_arm" || check_zone(M.zone_selected) == "l_arm")
+			H.visible_message( \
+				"<span class='notice'>[H] shakes [target]'s hand.</span>", \
+				"<span class='notice'>You shake [target]'s hand.</span>", )
+			
+		else if(check_zone(M.zone_selected) == "mouth") // I ADDED BOOP-EH-DEH-NOSEH - Jon
+			H.visible_message( \
```
Comment (deathride58) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
			M.visible_message( \
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 6]
```diff
@@ -316,7 +316,18 @@
...
+				"<span class='notice'>[H] boops [target]'s nose.</span>", \
```
Comment (deathride58) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
				"<span class='notice'>[M] boops [src]'s nose.</span>", \
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 7]
```diff
@@ -316,7 +316,18 @@
 
 			else
 				return
-
+		
+		else if(check_zone(M.zone_selected) == "r_arm" || check_zone(M.zone_selected) == "l_arm")
+			H.visible_message( \
+				"<span class='notice'>[H] shakes [target]'s hand.</span>", \
+				"<span class='notice'>You shake [target]'s hand.</span>", )
```
Comment (deathride58) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
				"<span class='notice'>You shake [src]'s hand.</span>", )
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 8]
```diff
@@ -316,7 +316,18 @@
...
+		
+		else if(check_zone(M.zone_selected) == "r_arm" || check_zone(M.zone_selected) == "l_arm")
+			H.visible_message( \
+				"<span class='notice'>[H] shakes [target]'s hand.</span>", \
+				"<span class='notice'>You shake [target]'s hand.</span>", )
+			
+		else if(check_zone(M.zone_selected) == "mouth") // I ADDED BOOP-EH-DEH-NOSEH - Jon
+			H.visible_message( \
+				"<span class='notice'>[H] boops [target]'s nose.</span>", \
+				"<span class='notice'>You boop [target] on the nose.</span>", )
```
Comment (deathride58) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
				"<span class='notice'>You boop [src] on the nose.</span>", )
```

**Review by deathride58 — CHANGES_REQUESTED** [review]
Just need to fix a few compiling errors now

**Comment by r4d6** [comment]
Done

**Review by deathride58 — APPROVED** [review]

**Pull request merged**